prov/gni/src/gnix_rma.c

```diff
@@ -740,6 +795,152 @@ int _gnix_rma_post_rdma_chain_req(void *data)
+	while (current != NULL) {
+		/* Get fab_req pointer */
+		more_req = container_of(current, struct gnix_fab_req, rma.sle);
+
+		/* Populate txd based on type */
+		if (req->type == GNIX_FAB_RQ_RDMA_WRITE) {
+			more_put[idx].ep_hndl = more_req->vc->gni_ep;
+			more_put[idx].length = more_req->rma.len;
+			more_put[idx].remote_addr = more_req->rma.rem_addr;
+			more_put[idx].local_addr = (uint64_t)more_req->
+								rma.loc_addr;
+
+			_gnix_convert_key_to_mhdl_no_crc(
+				(gnix_mr_key_t *)&more_req->rma.rem_mr_key,
+				&mdh);
+			more_put[idx].remote_mem_hndl = mdh;
+
+			if (idx < entries - 1)
+				more_put[idx].next_descr = &more_put[idx + 1];
+			else
+				more_put[idx].next_descr = NULL;
+		} else {
+			/* TODO Populate for reads*/
+			if (idx < entries - 1)
+				more_get[idx].next_descr = &more_get[idx + 1];
+			else
+				more_get[idx].next_descr = NULL;
+		}
+		idx++;
+		current = current->next;
+	}
+
+	if (req->type == GNIX_FAB_RQ_RDMA_WRITE) {
+		txd->gni_desc.next_descr = &more_put[0];
+	} else {
+		txd->gni_desc.next_descr = &more_get[0];
+	}
+}
+
+int _gnix_rma_more_post_req(void *data)
+{
+	struct gnix_fab_req *fab_req = (struct gnix_fab_req *)data;
+	struct gnix_fid_ep *ep = fab_req->gnix_ep;
+	struct gnix_nic *nic = ep->nic;
+	struct gnix_fid_mem_desc *loc_md;
+	struct gnix_tx_descriptor *txd;
+	gni_mem_handle_t mdh;
+	gni_return_t status;
+	int rc;
+
+	if (!gnix_ops_allowed(ep, fab_req->vc->peer_caps, fab_req->flags)) {
+		rc = __gnix_rma_post_err_no_retrans(fab_req, FI_EOPNOTSUPP);
+		if (rc != FI_SUCCESS)
+			GNIX_WARN(FI_LOG_EP_DATA,
+				  "__gnix_rma_post_err_no_retrans() failed: %d\n",
+				  rc);
+		return -FI_ECANCELED;
+	}
+
+	rc = _gnix_nic_tx_alloc(nic, &txd);
+	if (rc) {
+		GNIX_INFO(FI_LOG_EP_DATA,
+				"_gnix_nic_tx_alloc() failed: %d\n",
+				rc);
+		return -FI_ENOSPC;
+	}
+
+	txd->completer_fn = __gnix_rma_more_txd_complete;
+	txd->req = fab_req;
+
+	_gnix_convert_key_to_mhdl_no_crc(
+			(gnix_mr_key_t *)&fab_req->rma.rem_mr_key,
+			&mdh);
+
+	txd->gni_desc.type = __gnix_fr_post_type(fab_req->type, 0);
+	txd->gni_desc.cq_mode = GNI_CQMODE_GLOBAL_EVENT; /* check flags */
+	txd->gni_desc.dlvr_mode = GNI_DLVMODE_PERFORMANCE; /* check flags */
+
+	__gnix_rma_more_fill_pd(fab_req, txd);
+
+	if (fab_req->type == GNIX_FAB_RQ_RDMA_WRITE) {
+		GNIX_INFO(FI_LOG_EP_DATA, "MORE: Doing a Write\n");
+		txd->gni_desc.local_addr = (uint64_t)fab_req->rma.loc_addr;
+		txd->gni_desc.length = fab_req->rma.len;
+		assert(txd->gni_desc.length);
+		txd->gni_desc.remote_addr = (uint64_t)fab_req->rma.rem_addr;
+
+		loc_md = (struct gnix_fid_mem_desc *)fab_req->rma.loc_md;
+		if (loc_md) {
+			txd->gni_desc.local_mem_hndl = loc_md->mem_hndl;
+		}
+	}
+	txd->gni_desc.remote_mem_hndl = mdh;
+	txd->gni_desc.rdma_mode = 0; /* check flags */
+	txd->gni_desc.src_cq_hndl = nic->tx_cq; /* check flags */
+
+	GNIX_LOG_DUMP_TXD(txd);
+
+	COND_ACQUIRE(nic->requires_lock, &nic->lock);
+	GNIX_INFO(FI_LOG_EP_DATA, "MORE: Acquired Lock\n");
```

Comment:
Delete this debug statement or turn it into GNIX_DEBUG if you still think it might be useful.